pkg/exchange/coinbase/stream_handlers.go

```diff
@@ -0,0 +1,371 @@
+package coinbase
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+type channelType struct {
+	Name       string   `json:"name"`
+	ProductIDs []string `json:"product_ids,omitempty"`
+}
+
+func (c channelType) String() string {
+	return fmt.Sprintf(`(Name:       %s,
+ProductIDs: %s)`, c.Name, c.ProductIDs)
+}
+
+type authMsg struct {
+	Signature  string `json:"signature,omitempty"`
+	Key        string `json:"key,omitempty"`
+	Passphrase string `json:"passphrase,omitempty"`
+	Timestamp  string `json:"timestamp,omitempty"`
+}
+
+func (msg authMsg) String() string {
+	return fmt.Sprintf(`(Signature:  %s,
+Key:        ****,
+Passphrase: ****,
```

Comment:
We don't have to print all fields of the struct, so we could just ignore key and passphrase

Reply:
ok